modules/core/src/main/java/org/apache/ignite/internal/IgnitionEx.java

```diff
@@ -2623,28 +2617,44 @@ private static CacheConfiguration utilitySystemCache() {
          *
          * @param cancel Flag indicating whether all currently running jobs
          *      should be cancelled.
-         * @param waitForBackups Flag indicating whether sufficient backups on caches
-         *      should be waited for.
+         * @param shutdown This is a policy of shutdown which applied forcibly.
+         * If this property is null policy will apply policy of whole cluster.
          */
-        void stop(boolean cancel, boolean waitForBackups) {
+        void stop(boolean cancel, ShutdownPolicy shutdown) {
             // Stop cannot be called prior to start from public API,
             // since it checks for STARTED state. So, we can assert here.
             assert startGuard.get();
 
+            if (shutdown == null)
+                shutdown = determineShutdownPolicy();
+
             // If waiting for backups due to earlier invocation of stop(), stop wait and proceed shutting down.
-            if (!waitForBackups && waitingForBackups)
-                waitingForBackups = false;
+            if (shutdown == ShutdownPolicy.IMMEDIATE)
+                delayedShutdown = false;
 
-            stop0(cancel, waitForBackups);
+            stop0(shutdown, cancel);
         }
 
         /**
-         * @param cancel Flag indicating whether all currently running jobs
+         * Read policy from meta storage or return a default value if it didn't find.
+         *
+         * @return Shutdown policy.
+         */
+        private ShutdownPolicy determineShutdownPolicy() {
+            if (waitForBackups)
+                return ShutdownPolicy.GRACEFUL;
+
+            return grid.cluster().shutdownPolicy();
+        }
+
+        /**
+         * Stop instance synchronously according to parameters.
+         *
+         * @param shutdown Policy of according which shutdown will be produced.
+         * @param cancel Flag indicating whether all currently running jobsCorruptedCheckpointReservationTest
          *      should be cancelled.
-         * @param waitForBackups Flag indicating whether sufficient backups on caches
-         *      should be waited for.
          */
-        private synchronized void stop0(boolean cancel, boolean waitForBackups) {
+        private synchronized void stop0(ShutdownPolicy shutdown, boolean cancel) {
```

Comment:
Minor: seems that almost everywhere we have "boolean cancel" as first parameter.